File: 2_Programação_sistemas_operacionais/2.1_File_stdio/Python/Ex7.py
import struct

def abre_arq(arquivo, modo):
	try:
		p = open(arquivo, modo)
	except OSError:
		print("Erro! Impossivel abrir o arquivo!")
		exit(-1)
	return p

pi = float(3.1415)
fp = abre_arq("arquivo.bin", "wb")
fp.write(struct.pack('f',pi))
fp.close()
fp = abre_arq("arquivo.bin", "rb")
pilido = struct.unpack('f',fp.read(4))
print("O valor de PI (lido do arquivo) é", pilido[0])
fp.close()

File: 2_Programação_sistemas_operacionais/2.1_File_stdio/Ex7.c
#include <stdio.h>
#include <stdlib.h>
#include "file_stdio.h"

int main()
{
	FILE *fp;
	float pi = 3.1415;
	float pilido;

	fp = abre_arq("arquivo.bin", "wb");
	if(fwrite(&pi, sizeof(float), 1, fp) != 1)
		printf("Erro na escrita do arquivo");
	fclose(fp);

	fp = abre_arq("arquivo.bin", "rb");
	if(fread(&pilido, sizeof(float), 1, fp) != 1)
		printf("Erro na leitura do arquivo");
	printf("O valor de PI "
		"(lido do arquivo) eh: %f\n",
		pilido);
	fclose(fp);
	return(0);
}
